v1/bundle/store.go

```diff
@@ -827,7 +827,7 @@ func writeModuleRegoVersionToStore(ctx context.Context, store storage.Store, txn
 
 	if regoVersion == ast.RegoUndefined {
 		var err error
-		regoVersion, err = b.RegoVersionForFile(mf.Path, ast.RegoUndefined)
```

Comment:
This was the root cause of the issue. If no rego-version was declared for the module in the bundle, we'd always default to `v1`, and not the runtime rego-version (i.e. `--v0-compatible` enabled/disabled).